### ISSUE TITLE
Rule names inconsistent with the standard ESLint ruleset

### ISSUE THREAD

**Issue by remcohaszing** [Gillespie59/eslint-plugin-angular#177]
The standard ESLint ruleset uses hyphens for the rule names. This plugin uses underscores. It might be better to stick to whatever ESLint uses, in order to keep more consistent `.eslintrc` files.

While at the issue of rule naming conventions; each rule uses the `ng_` prefix. In a `.eslintrc` file this results in prefix of `angular/ng_`.

The result this `.eslintrc` snippet:
```yaml
rules:
  - no-extra-semi: 2
  - angular/ng_document_service: 2
```

I propose to use hyphens and remove the `ng_` prefix. The `.eslintrc` snippet would then look like this:
```yaml
rules:
  - no-extra-semi: 2
  - angular/document-service: 2
```

The latter seems cleaner to me.

**Comment by Gillespie59** [comment]
Good idea. But how we can avoid breaking every ESLint configuration file using our plugin ?

**Comment by remcohaszing** [comment]
We can't without a hacky solution. I think this should be part of version 1.0.0. Bumping the major version may contain breaking changes conforming to [semver](http://semver.org).

I don't think this is a high priority issue, but when bumping to 1.0.0 for whatever reason could include this patch as well.

**Comment by tilmanpotthof** [comment]
As long as the version is 0.x breaking changes are ok. How about a workaround (until 1.0) which allows both configuration styles, but the "old" style log a deprecation warning.

**Comment by Gillespie59** [comment]
What about create a deprecated folder with the existing rules (with the underscore), in which we add a warning each time the rule is executed. 

An in the current directory, a copy/past of each rule but with a "-"

Both directory will be injected in the index.js file. 

By default, all rules with the underscore will be disabled.

**Comment by tilmanpotthof** [comment]
I wrote an abstraction for the rule configuration which has the same effect. Just without copying the rules. The documentation is not updated.

https://github.com/Gillespie59/eslint-plugin-angular/pull/187

**Comment by Gillespie59** [comment]
THanks a lot @tilmanpotthof 

I have merged everything. If everything is OK, I will publish the 0.10.0 release this week-end. 

Manu
